Sources/Honeycomb/CompositeSpanProcessor.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+import OpenTelemetryApi
+import OpenTelemetrySdk
+
+internal class CompositeSpanProcessor: SpanProcessor {
+    private var spanProcessors: [SpanProcessor] = []
+    var isStartRequired: Bool = false
+    var isEndRequired: Bool = false
+
+    func addSpanProcessor(_ spanProcessor: SpanProcessor) {
+        spanProcessors.append(spanProcessor)
+        isStartRequired = spanProcessors.contains(where: { $0.isStartRequired })
```

Comment:
Would we need to do this over all the span processors everytime? does the "start/end required" change for already added ones?

Reply:
Theoretically, I guess it's possible for `isStartRequired`/`isEndRequired` to change at runtime, since we don't know what SpanProcessors we'll be receiving...

so maybe it makes more sense for these to become computed properties? Like so:

```swift
var isStartRequired: Bool {
    spanProcessors.contains(where: { $0.isStartRequired })
}
var isEndRequired: Bool {
    spanProcessors.contains(where: { $0.isEndRequired })
}
```

Reply:
ok I thought about it a bit more and I'm going to take the computed properties approach!